app/models/flag_condition.rb

```diff
@@ -65,10 +65,14 @@ def self.validate_for_user(user, disabler)
     accuracy
   end
 
-  def validate!(post)
-    post.reasons.pluck(:weight).reduce(:+) >= min_weight &&
-      post.stack_exchange_user.reputation <= max_poster_rep &&
-      post.reasons.count >= min_reason_count
+  def validate_weight_rep_reason_count!(weight, poster_rep, reason_count)
+    weight >= min_weight && poster_rep <= max_poster_rep && reason_count >= min_reason_count
+  end
+
+  def validate_post!(post)
```

Comment:
Might be a good idea to leave this one called `validate!` - just in case there are any other places in the code where `validate!` is still used.

Reply:
I'm happy to change that back. I *should have* mentioned that prior to changing it, I did a `grep "validate!" -R` on the repository. The only results were:

```
app/models/concerns/post_concerns/autoflagging.rb:23:          if condition.validate!(post)
app/models/concerns/post_concerns/autoflagging.rb:166:        if condition.validate!(self)
app/models/flag_condition.rb:68:  def validate!(post)
```

I'll change it back.